shared/src/balancer/event_handler.rs

```diff
@@ -32,27 +38,36 @@ pub struct PoolCreated {
     pub pool_address: H160,
 }
 
-#[derive(Clone, Debug, Default, Eq, PartialEq, Hash)]
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
+pub enum PoolType {
+    WeightedGeneral,
+    WeightedTwoToken,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
```

Comment:
These hashes are all used because we have `RegisteredWeightedPool` as a value in a HashMap.

Reply:
`Hash` is only needed if it used as a key, not a value.

Reply:
```
error[E0277]: the trait bound `RegisteredWeightedPool: std::hash::Hash` is not satisfied
   --> shared/src/balancer/event_handler.rs:133:14
    |
133 |             .collect::<HashSet<RegisteredWeightedPool>>()
    |              ^^^^^^^ the trait `std::hash::Hash` is not implemented for `RegisteredWeightedPool`
    |
    = note: required because of the requirements on the impl of `FromIterator<RegisteredWeightedPool>` for `std::collections::HashSet<RegisteredWeightedPool>`

```